### PR TITLE
Bcu 8#suport kubernetes engine

### PR DESCRIPTION
**Pull request by tung491**

### CODE REVIEW

File: cmd/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+		}
+		formatter.Output(kubernetesClusterHeader, data)
+	}}
+
+var clusterCreate = &cobra.Command{
+	Use:   "create",
+	Short: "Create Kubernetes cluster with worker pool",
+	Long: `Create Kubernetes cluster with worker pool using file or flags (Sample config file in example)
+- Using flag example: ./bizflycloud kubernetes create --name test_cli --version 5f7d3a91d857155ad4993a32 --vpc-network-id 145bed1f-a7f7-4f88-ab3d-ce2fc95a4e71 -tag abc -tag xyz --worker-pool name=testworkerpool,flavor=nix.3c_6g,profile_type=premium,volume_type=PREMIUM-HDD1,volume_size=40,availability_zone=HN1,desired_size=1,min_size=1,max_size=10
+- Using config file example: ./bizflycloud kubernetes create --config-file create_cluster.json`,
```
Comment (ghost) [inline cmd/kubernetes.go]: use yaml for easier reading and editing

File: cmd/kubernetes.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			log.Fatal("Invalid arguments")
+		}
+		client, ctx := getApiClient(cmd)
+		err := client.KubernetesEngine.Delete(ctx, args[0])
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Printf("Delete the Kubernetes cluster successfully")
```
Comment (pmint93) [inline cmd/kubernetes.go]: Cluster is in the process of being deleted

File: cmd/kubernetes.go [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+var recycleNode = &cobra.Command{
+	Use:   "recycle-node",
+	Short: "Recycle Node",
+	Run: func(cmd *cobra.Command, args []string) {
+		client, ctx := getApiClient(cmd)
+		err := client.KubernetesEngine.RecycleNode(ctx, args[0], args[1], args[2])
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Recycling node successfully")
```
Comment (pmint93) [inline cmd/kubernetes.go:242]: Node is recycling now

File: cmd/kubernetes.go [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+var deleteWorkerPool = &cobra.Command{
+	Use:   "delete-workerpool",
+	Short: "Delete worker pool",
+	Run: func(cmd *cobra.Command, args []string) {
+		client, ctx := getApiClient(cmd)
+		err := client.KubernetesEngine.DeleteClusterWorkerPool(ctx, args[0], args[1])
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Delete worker pool successfully")
```
Comment (pmint93) [inline cmd/kubernetes.go]: Worker pool is in the process of being deleted

File: cmd/kubernetes.go [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+			DesiredSize:       desiredSize,
+			EnableAutoScaling: enableAutoScaling,
+			MinSize:           minSize,
+			MaxSize:           maxSize,
+		}
+		err := client.KubernetesEngine.UpdateClusterWorkerPool(ctx, args[0], args[1], uwr)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Update worker pool successfully")
```
Comment (pmint93) [inline cmd/kubernetes.go]: Worker pool is updating now

File: cmd/kubernetes.go [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+var deleteWorkerPoolNode = &cobra.Command{
+	Use:   "delete-node",
+	Short: "Delete node",
+	Run: func(cmd *cobra.Command, args []string) {
+		client, ctx := getApiClient(cmd)
+		err := client.KubernetesEngine.DeleteClusterWorkerPoolNode(ctx, args[0], args[1], args[2])
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Delete worker pool node successfully")
```
Comment (pmint93) [inline cmd/kubernetes.go]: Node successfully deleted

File: cmd/kubernetes.go [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+	kubernetesCmd.AddCommand(addWorkerPool)
+
+	uwp := updateWorkerPool.PersistentFlags()
+	uwp.IntVar(&desiredSize, "desired-size", 0, "Desired size")
+	uwp.BoolVar(&enableAutoScaling, "autoscaling", false, "Enable Auto scaling")
+	uwp.IntVar(&minSize, "min-size", 0, "Min size")
+	uwp.IntVar(&maxSize, "max-size", 0, "Max size")
+	kubernetesCmd.AddCommand(updateWorkerPool)
+
+	getKubeConfig.PersistentFlags().StringVar(&outputKubeConfigFilePath, "output-path", ".", "Output path")
```
Comment (pmint93) [inline cmd/kubernetes.go]: should be --output or -o

File: cmd/kubernetes.go [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+		}
+		fmt.Printf("Cluster is in the process of being deleted")
+	},
+}
+
+var addWorkerPool = &cobra.Command{
+	Use:   "add",
+	Short: "Add worker pool into cluster",
+	Long: `Add Kubernetes worker pool using file or flags (Sample config file in example)
+- Using flag example: ./bizfly kubernetes add-workerpool xfbxsws38dcs8o94 --worker-pool name=testworkerpool,flavor=nix.3c_6g,profile_type=premium,volume_type=PREMIUM-HDD1,volume_size=40,availability_zone=HN1,desired_size=1,min_size=1,max_size=10
```
Comment (ghost) [inline cmd/kubernetes.go]: please fix the example ?

File: cmd/kubernetes.go [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+		err := client.KubernetesEngine.DeleteClusterWorkerPoolNode(ctx, args[0], args[1], args[2])
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println("Worker pool is in the process of being deleted")
+	},
+}
+
+var getKubeConfig = &cobra.Command{
+	Use:   "get-kubeconfig",
```
Comment (ghost) [inline cmd/kubernetes.go]: I think we should not use `-` in the command line.

File: cmd/kubernetes.go [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+	Use:   "kubernetes",
+	Short: "Bizfly Kubernetes Engine Interaction",
+	Long:  "Bizfly Kubernetes Engine Action: List, Create, Delete, Get",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("kubernetes engine called")
+	},
+}
+
+var kubernetesWorkerPoolCmd = &cobra.Command{
+	Use:   "worker-pool",
```
Comment (ghost) [inline cmd/kubernetes.go]: use workerpool instead

File: cmd/kubernetes.go (hunk 9 above)
Comment (ghost) [reply cmd/kubernetes.go]: how about `bizfly kubernetes kubeconfig get <cluster id>`

**Review by ghost — CHANGES_REQUESTED** [review]

**Pull request merged**